tests/unit/conftest.py

```diff
@@ -3,27 +3,37 @@
 from unittest.mock import MagicMock, patch
 
 import pytest
-from ops.testing import Context, Exec
+from ops.testing import Context
 
 from charm import OpenTelemetryCollectorCharm
 
+CHARM_ROOT=Path(__file__).parent.parent.parent
 
 @pytest.fixture
-def ctx(tmp_path):
+def unit_id():
+    return 0
+
+@pytest.fixture
+def app_name():
+    return "otelcol"
+
+
+@pytest.fixture
+def unit_name(unit_id, app_name):
+    return f"{app_name}/{unit_id}"
+
+
+@pytest.fixture
+def ctx(tmp_path, unit_id, app_name):
     src_dirs = ["grafana_dashboards", "loki_alert_rules", "prometheus_alert_rules"]
     # Create a virtual charm_root so Scenario respects the `src_dirs`
     # Related to https://github.com/canonical/operator/issues/1673
     for src_dir in src_dirs:
-        source_path = Path("src") / src_dir
+        source_path = CHARM_ROOT/ "src" / src_dir
         target_path = tmp_path / "src" / src_dir
         copytree(source_path, target_path, dirs_exist_ok=True)
     with patch("charm.refresh_certs", lambda: True):
-        yield Context(OpenTelemetryCollectorCharm, charm_root=tmp_path)
-
-
-@pytest.fixture
-def execs():
-    yield {Exec(["update-ca-certificates", "--fresh"], return_code=0, stdout="")}
```

Comment:
Why the removal?
We don't want this to run on the test host. (Not even if it's in `spread`.)